docs/classic-ui/mockup.md

```diff
@@ -52,6 +52,80 @@ You can check the full list of available features using the `-l` parameter:
 plonecli -l
 ```
 
+## Create a custom pattern
+
+To create a custom {term}`pattern` in your add-on, use the `mockup_pattern` {term}`bobtemplate`:
+
+```shell
+cd project.addon
+plonecli add mockup_pattern
+```
+
+Next, enter your pattern name without the `pat-` prefix.
+
+```shell
+--> Pattern name (without “pat-” prefix) [my-pattern]: testpattern
+```
+
+This creates the necessary JavaScript resources and webpack configuration for you, as shown in the following file system tree diagram.
+
+```text
+...
+├── resources
+|   ├── pat-testpattern
+|   |   ├── documentation.md
+|   |   ├── testpattern.js
+|   |   ├── testpattern.scss
+|   |   ├── testpattern.test.js
+│   ├── bundle.js
+│   ├── index.html
+│   ├── index.js
+├── package.json
+├── webpack.config.js
+...
+```
+
+All your pattern JavaScript code goes into {file}`resources/pat-testpattern/testpattern.js`.
+SCSS files can be imported, too, since webpack provides the `sass-loader` module.
+
+Next, install the npm packages using {term}`yarn`.
+
+```shell
+yarn install
+```
+
+When you finish writing your JavaScript code, you have to build the bundle with the following command.
+
+```shell
+yarn build
+```
+
+This creates the webpack chunks and the JavaScript bundle files in your add-on package.
+
+```text
+...
+├── src
+|   ├── project
+|   |   ├── addon
+|   |   |   ├── browser
+|   |   |   |   ├── static
+|   |   |   |   |   ├── bundles
+|   |   |   |   |   |   ├── chunks
+|   |   |   |   |   |   ├── addon-remote.min.js
+|   |   |   |   |   |   ├── addon-remote.min.js.map
+|   |   |   |   |   |   ├── addon.min.js
+|   |   |   |   |   |   ├── addon.min.js.map
+```
+
+Note that `plonecli` also creates an XML file in {file}`src/project/addon/profiles/default/registry/bundles.xml`, which registers the {file}`addon-remote.min.js` file in the resource registry.
+
+```{important}
+You must re-import your profile with an upgrade step if you installed your add-on in Plone before adding the pattern.
```

Comment:
How does one do this step?

Reply:
Simplest way: uninstall/install the addon

I searched for the genericsetup documentation and got a huge old document here https://5.docs.plone.org/develop/addons/components/genericsetup.html#upgrade-steps which is still 85% valid ... would be goot to polish it and take it over to the actual docs.

Reply:
Yeah, GenericSetup docs have not been ported to Plone 6 yet. I don't know which parts are valid or not. We can do a separate PR for them. I can convert this file https://github.com/plone/documentation/blob/5.2/develop/addons/components/genericsetup.rst from .rst to .md, then you can fiddle with it. Where should its content go? I think most of it is how-to guide.

Reply:
That would be great. Best place would be inside a "howto create a backend addon" guide

Reply:
I created a new issue to track this at https://github.com/plone/documentation/issues/1754

Reply:
PR started in https://github.com/plone/documentation/pull/1757